core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -66,6 +68,8 @@ class LogSegment(val log: FileRecords,
   /* The maximum timestamp we see so far */
   @volatile private var maxTimestampSoFar = timeIndex.lastEntry.timestamp
   @volatile private var offsetOfMaxTimestamp = timeIndex.lastEntry.offset
+  /* The offset that would be used for the next message to be appended to this segment */
+  @volatile private var _nextOffset = INVALID_NEXT_OFFSET
```

Comment:
To be consistent with the logic in recovery, should we initialize this to baseOffset?